app/views/events/index.html.erb

```diff
@@ -0,0 +1,19 @@
+<h1>All events</h1>
+<p>Find me in app/views/events/index.html.erb</p>
+
+
+Upcoming Events:<br>
+<% @upcoming_events.each do |e| %>
+  <%= e.title %><br>
+  <%= e.date %><br>
+  <%= link_to "Click to see the event", event_path(e) %>
+<% end %>
+
+<br>
+
+Past Events:<br>
+<% @past_events.each do |e| %>
+  <%= e.title %><br>
+  <%= e.date %><br>
+  <%= link_to "Click to see the event", event_path(e) %>
+<% end %>
```

Comment:
If you had that partial (shared in this case), it would have saved you the repetitions.